test/features/errors/notfound.js

```diff
@@ -66,4 +70,95 @@ describe('404 handling', function() {
+            .post('')
+            .reply(200, {
+                'batchcomplete': '',
+                'query': {
+                    'pages': {
+                        '11089416': {
+                            'pageid': 11089416,
+                            'ns': 0,
+                            'title': title,
+                            'contentmodel': 'wikitext',
+                            'pagelanguage': 'en',
+                            'touched': '2015-05-22T08:49:39Z',
+                            'lastrevid': 653508365,
+                            'length': 2941,
+                            'revisions': [{
+                                'revid': revision,
+                                'user': 'Chuck Norris',
+                                'userid': 3606755,
+                                'timestamp': '2015-03-25T20:29:50Z',
+                                'size': 2941,
+                                'sha1': 'c47571122e00f28402d2a1b75cff77a22e7bfecd',
+                                'contentmodel': 'wikitext',
+                                'comment': 'Test',
+                                'tags': []
+                            }]
+                        }
+                    }
+                }
+            })
+            // Other requests return nothing as if the page is deleted.
+            .post('')
+            .reply(200, {})
+            .post('')
+            .reply(200, {});
+        // Fetch the page
+        return preq.get({ uri: server.config.bucketURL + '/title/' + title + '/latest'})
+            .then(function(res) {
+                assert.deepEqual(res.status, 200);
+                assert.deepEqual(res.body.items.length, 1);
+                assert.deepEqual(res.body.items[0].rev, revision);
+            })
+            // Now fetch info that it's deleted
+            .then(function() {
+                return preq.get({ uri: server.config.bucketURL + '/title/' + title + '/latest'});
+            })
+            .then(function() {
+                throw new Error('404 should have been returned for a deleted page');
+            })
+            .catch(function(e) {
+                assert.deepEqual(e.status, 404);
+                assert.contentType(e, 'application/problem+json');
+            })
+            // Getting it by revision id should also return 404
+            .then(function() {
+                return preq.get({ uri: server.config.bucketURL + '/revision/' + revision});
+            })
+            .then(function() {
+                throw new Error('404 should have been returned for a deleted page');
+            })
+            .catch(function(e) {
+                assert.deepEqual(e.status, 404);
+                assert.contentType(e, 'application/problem+json');
+            })
+            .then(function() {
+                api.done();
+            })
+            .finally(function() {
+                nock.cleanAll();
+                nock.restore();
+            });
```

Comment:
@gwicke It is checked and disabled. Look to these lines: done() verifies that every call we've intercepted was actually called, and cleanAll+restore cleans nock context and restores http to the original stated (disables nock). 

I've tested that failure of this test doesn't make other tests fail.